src/v/datalake/record_schema_resolver.cc

```diff
@@ -96,7 +96,7 @@ struct schema_translating_visitor {
             auto type = type_to_iceberg(*d).value();
             co_return type_and_buf{
               .type = resolved_type{
-                .schema = *d,
+                .schema = wrapped_protobuf_descriptor { *d, pb_def },
```

Comment:
Can pb_def be moved?

Reply:
It's possible, `pb_def` is just a wrapper around a `shared_ptr` to the underlying descriptor pool though so it won't change much. I'll switch to moving it though in case the `shared_ptr` is ever switched to something else.